src/s3_app.erl

```diff
@@ -0,0 +1,42 @@
+%%% @doc
+%%% Application module for S3 client
+%%% @end
+-module(s3_app).
+
+-behaviour(application).
+
+%% Application callbacks
+-export([start/0, start/2, stop/0, stop/1]).
+
+%% API
+start() ->
+    application:ensure_all_started(s3erl).
```

Comment:
Wouldn't `application:start/1` suffice here? Dependencies should have been already started by now.